drivers/i2s/i2s_ll_stm32.c

```diff
@@ -529,6 +574,16 @@ static void dma_rx_callback(const struct device *dma_dev, void *arg,
 
 	mblk_tmp = stream->mem_block;
 
+	if ((stream->state == I2S_STATE_STOPPING) && (status == DMA_STATUS_COMPLETE)) {
+		if (stream->tx_stop_for_drain == false) {
```

Comment:
There is no need to check this on RX because `I2S_TRIGGER_DRAIN` is redirected to `I2S_TRIGGER_STOP`. Ignoring it also removes the need to update it in `i2s_stm32_trigger`.